<a href="https://colab.research.google.com/github/anabiarochar/Mestrado_AlgoritmosProgramacao/blob/main/Projeto1/Bolha.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [38]:
import requests
import time

# 1. Obter os dados do GitHub
# Nota: Lembre-se que o token no URL pode expirar.
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/refs/heads/main/Projeto1/Vetores/vetor_100000_aleatorio.txt?token=GHSAT0AAAAAADXG3NI5QVAGZITGMBNKBX3A2NNGIEA"

try:
    response = requests.get(url)
    content = response.text

    # AJUSTE AQUI: Separando por vírgula e convertendo para inteiro
    # Usamos .strip() para remover possíveis espaços em branco ao redor dos números
    vetor = [int(x.strip()) for x in content.split(',') if x.strip()]

    print(f"Sucesso! Vetor carregado com {len(vetor)} elementos.")
    print(f"Primeiros 5 elementos lidos: {vetor[:5]}")

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")
    vetor = []

# 2. Implementação do Método Bolha (Bubble Sort)
def bubble_sort(lista_original):
    # Criamos uma cópia para preservar o vetor original
    arr = lista_original.copy()
    n = len(arr)

    for i in range(n):
        trocou = False
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                # Troca os elementos de posição
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                trocou = True

        # Se não houve troca nesta iteração, o vetor já está ordenado
        if not trocou:
            break
    return arr

# 3. Execução e Medição de Tempo
if vetor:
    print("\nIniciando ordenação por Bubble Sort...")

    # Início do cronômetro
    inicio = time.perf_counter()

    vetor_ordenado = bubble_sort(vetor)

    # Fim do cronômetro
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print("--- Resultado ---")
    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(f"Primeiros 10 elementos ordenados: {vetor_ordenado[:10]}")
    print(f"Últimos 10 elementos ordenados: {vetor_ordenado[-10:]}")
else:
    print("O vetor está vazio. Verifique o link ou o formato do arquivo.")

Sucesso! Vetor carregado com 100000 elementos.
Primeiros 5 elementos lidos: [33462, 35119, 52322, 11959, 15234]

Iniciando ordenação por Bubble Sort...
--- Resultado ---
Tempo de execução: 735.920505 segundos
Primeiros 10 elementos ordenados: [2, 5, 6, 6, 8, 9, 10, 13, 14, 14]
Últimos 10 elementos ordenados: [99989, 99990, 99990, 99991, 99992, 99992, 99997, 99998, 99998, 100000]


###Aleatório 1000:
Tentativa 1 - Tempo de execução: 0.078169 segundos  
Tentativa 2 - Tempo de execução: 0.050635 segundos  
Tentativa 3 - Tempo de execução: 0.095585 segundos  
Tentativa 4 - Tempo de execução: 0.051358 segundos  

###Aleatório 10000:
Tentativa 1 - Tempo de execução: 7.173240 segundos  
Tentativa 2 - Tempo de execução: 7.225697 segundos  
Tentativa 3 - Tempo de execução: 7.230197 segundos  
Tentativa 4 - Tempo de execução: 6.075342 segundos  

###Aleatório 100000:
Tentativa 1 - Tempo de execução: 776.446981 segundos  
Tentativa 2 - Tempo de execução: 765.963112 segundos  
Tentativa 3 - Tempo de execução: 735.920505 segundos    
Tentativa 4 - Tempo de execução:  

###Crescente 1000:
Tentativa 1 - Tempo de execução: 0.003487 segundos  
Tentativa 2 - Tempo de execução: 0.000092 segundos  
Tentativa 3 - Tempo de execução: 0.000088 segundos  
Tentativa 4 - Tempo de execução: 0.000081 segundos

###Crescente 10000:
Tentativa 1 - Tempo de execução: 0.000803 segundos  
Tentativa 2 - Tempo de execução: 0.000956 segundos  
Tentativa 3 - Tempo de execução: 0.001006 segundos  
Tentativa 4 - Tempo de execução: 0.000927 segundos  

###Crescente 100000:
Tentativa 1 - Tempo de execução: 0.013913 segundos  
Tentativa 2 - Tempo de execução: 0.009523 segundos  
Tentativa 3 - Tempo de execução: 0.009411 segundos  
Tentativa 4 - Tempo de execução: 0.012313 segundos  

###Decrescente 1000:
Tentativa 1 - Tempo de execução: 0.063765 segundos  
Tentativa 2 - Tempo de execução: 0.062513 segundos  
Tentativa 3 - Tempo de execução: 0.061670 segundos  
Tentativa 4 - Tempo de execução: 0.063310 segundos  

###Decrescente 10000:
Tentativa 1 - Tempo de execução: 8.937898 segundos  
Tentativa 2 - Tempo de execução: 8.585050 segundos  
Tentativa 3 - Tempo de execução: 8.502795 segundos  
Tentativa 4 - Tempo de execução: 8.747462 segundos  

###Decrescente 100000:
Tentativa 1 - Tempo de execução: 858.927049 segundos  
Tentativa 2 - Tempo de execução: 858.957241 segundos  
Tentativa 3 - Tempo de execução: 855.807456 segundos   
Tentativa 4 - Tempo de execução:   

